tests/test_dataset.py

```diff
@@ -463,3 +463,50 @@ def test_dataset_current_records_index_filtering_accurate_records_yielded(
         "alma:23",
         "alma:24",
     ]
+
+
+@pytest.mark.freeze_time("2025-05-22 01:23:45.567890")
+def test_dataset_write_includes_minted_run_timestamp(
+    dataset_with_same_day_runs,
+):
+    # assert TIMDEXDataset.write() applies current time as run_timestamp
+    row_dict = next(dataset_with_same_day_runs.read_dicts_iter())
+    assert "run_timestamp" in row_dict
+    assert row_dict["run_timestamp"] == datetime(
```

Comment:
Totally optional: you could use `strftime()` so it's immediately obvious that it matches `@pytest.mark.freeze_time("2025-05-22 01:23:45.567890")`